### PR TITLE
Validation exception stacktrace shortened

### LINKED ISSUE #140: Cut exception stack straces thrown in validation functions
When a validation function (annotated with `@Validate`) throws an exception, the full stack trace, including the Lincheck internals, is provided. It takes dozens of lines in the console, being useless at the same time. The suggestion is to cut the stack trace at the point of the validation function call.

### PR DESCRIPTION
**Pull request by avpotapov00**
Fixes #140

### CODE REVIEW

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt [hunk 1]
```diff
@@ -81,7 +82,15 @@ private fun executeValidationFunction(instance: Any, validationFunction: Method)
     try {
         m.invoke(instance)
     } catch (e: Throwable) {
-        return e.cause
+        // cut off lincheck-related and reflection-related stacktrace elements
+        val validationException = e.cause ?: return e
+        val wrapperExceptionStackTraceLength = e.stackTrace.size
+
+        if (e is InvocationTargetException) {
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt:89]: Can it be not an `InvocationTargetException`?

Comment (ndkoval) [reply src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt:89]: Please add a comment explaining when another exception type can be thrown, clarifying why simply returning it is correct. Maybe, we need to throw an internal exception instead.

**Review by ndkoval — CHANGES_REQUESTED** [review]
Please address the comment in `Utils.kt`

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt (hunk 1 above)
Comment (avpotapov00) [reply src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt]: Done

**Review by ndkoval — APPROVED** [review]

**Pull request merged**